app/uk/gov/hmrc/teamsandrepositories/config/JenkinsConfig.scala

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmrc.teamsandrepositories.config
+
+import javax.inject.Inject
+import play.api.Configuration
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+import scala.concurrent.duration.FiniteDuration
+
+class JenkinsConfig @Inject()(config: Configuration) {
+  lazy val username: String = config.getOptional[String]("jenkins.username").getOrElse(
+    throw new RuntimeException("Error getting config value jenkins.username"))
+
+  lazy val password: String = config.getOptional[String]("jenkins.password").getOrElse(
+    throw new RuntimeException("Error getting config value jenkins.password"))
+
+  lazy val baseUrl: String = config.getOptional[String]("jenkins.url").getOrElse(
+    throw new RuntimeException("Error getting config value jenkins.url")
+  )
+
+  private val defaultTimeout = 2 hours
+
+  def getDuration(key: String): Option[FiniteDuration] =
+    config.getOptional[FiniteDuration](key)
+
+  def reloadEnabled: Boolean = config.getOptional[Boolean]("jenkins.reloadEnabled").getOrElse(false)
```

Comment:
move default values into application.conf and make these fail if not found.
(one place for config values, and will show up in ConfigExplorer)